transport/s3-transport/src/main/java/org/apache/airavata/mft/transport/s3/S3SourceConnector.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *   Licensed to the Apache Software Foundation (ASF) under one
+ *   or more contributor license agreements.  See the NOTICE file
+ *   distributed with this work for additional information
+ *   regarding copyright ownership.  The ASF licenses this file
+ *   to you under the Apache License, Version 2.0 (the
+ *   "License"); you may not use this file except in compliance
+ *   with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ */
+
+package org.apache.airavata.mft.transport.s3;
+
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.S3Object;
+import org.apache.airavata.mft.core.api.ConnectorChannel;
+import org.apache.airavata.mft.core.api.SourceConnector;
+import org.apache.airavata.mft.core.bufferedImpl.AbstractConnector;
+import org.apache.airavata.mft.core.bufferedImpl.ConnectorConfig;
+import org.apache.airavata.mft.core.bufferedImpl.Constants;
+import org.apache.airavata.mft.core.bufferedImpl.InChannel;
+
+import java.io.InputStream;
+import java.util.Properties;
+
+/**
+ * Connector class which connects to a given S3 source and provides
+ * S3 Input Stream.
+ */
+public class S3SourceConnector extends AbstractConnector implements SourceConnector {
+
+    private AmazonS3 s3Client;
+
+    @Override
+    public boolean initiate(ConnectorConfig connectorConfig) {
```

Comment:
I understand that you are trying to make the APIs generic as possible. But in some cases, it might cause the API to less readable. Fro an outsider, it would be hard to understand what are the properties that he should pass into connectorConfig as it contains generic key value pairs. Instead of using that one, why don't we come up configs specific for each connector with clearly defined properties. Say S3ConnectorConfig. Then the user exactly knows what to pass. That will reduce the reusability a bit but will improve the clarity of the API. What do you think?

Reply:
Yes I agree. What I thought was  Connector implementation is specific to its transport attributes and local to that. Hence, anyone who tries to read code should know the transport specific attributes. So I thought it won't be a problem. But it makes Connector decouple from implementation details.